apps/client/src/common/utils/linkUtils.js

```diff
@@ -22,9 +22,15 @@ export function openLink(url) {
  * Handles opening external links
  * @param event
  * @param location
+ * @param alias (this is optional)
  */
-export function handleLinks(event, location) {
+export function handleLinks(event, location, alias = '') {
+  let appendAlias = '';
+  const connector = location.includes('?') ? '&' : '?';
+  if (alias && alias !== '') {
+    appendAlias = `${connector}alias=${alias}`;
+  }
   // we handle the link manually
   event.preventDefault();
-  openLink(`http://${host}/${location}`);
+  openLink(`http://${host}${location}${appendAlias}`);
 }
```

Comment:
I dont think this should be done here. 
Since this is an implementation detail of the alias feature, we would prefer adding this to the code responsible for this (the useEffect you had). I believe that would provide better encapsulation and help future code maintenance and development